src/mlpack/methods/ann/layer/convolution.hpp

```diff
@@ -176,135 +166,88 @@ class Convolution
    * @param gy The backpropagated error.
    * @param g The calculated gradient.
    */
-  template<typename eT>
-  void Backward(const arma::Mat<eT>& /* input */,
-                const arma::Mat<eT>& gy,
-                arma::Mat<eT>& g);
+  void Backward(const MatType& /* input */,
+                const MatType& gy,
+                MatType& g);
 
-  /*
+  /**
    * Calculate the gradient using the output delta and the input activation.
    *
    * @param input The input parameter used for calculating the gradient.
    * @param error The calculated error.
    * @param gradient The calculated gradient.
    */
-  template<typename eT>
-  void Gradient(const arma::Mat<eT>& /* input */,
-                const arma::Mat<eT>& error,
-                arma::Mat<eT>& gradient);
+  void Gradient(const MatType& /* input */,
+                const MatType& error,
+                MatType& gradient);
 
   //! Get the parameters.
-  OutputDataType const& Parameters() const { return weights; }
+  MatType const& Parameters() const { return weights; }
   //! Modify the parameters.
-  OutputDataType& Parameters() { return weights; }
+  MatType& Parameters() { return weights; }
 
-  //! Get the weight of the layer.
+  //! Get the weight of the layer as a cube.
```

Comment:
I can't comment on the line below but should this be `arma::Cube<typename  MatType::elem_type>`?

Reply:
Ahh, nice catch, fixed in 1a9f7ccd5.